src/gui/addnewtorrentdialog.cpp

```diff
@@ -94,6 +95,8 @@ AddNewTorrentDialog::AddNewTorrentDialog(QWidget *parent)
     connect(ui->browseButton, SIGNAL(clicked()), SLOT(browseButton_clicked()));
     ui->defaultSavePathCheckBox->setVisible(false); // Default path is selected by default
 
+    ui->doNotDeleteTorrentCheckBox->setVisible(TorrentFileGuard::deleteAddedTorrents());
```

Comment:
>Actually "Delete .torrent file afterwards" should be a GroupBox and "Also when addition is cancelled" should be a child of it. And the warning label too. This is what we do with other options in that dialog.

Scratch that. You already do this. So just delete the above. I don't hiding options in that dialog. The GroupBox should take care of enabling/disabling its children already.